src/views/product-landing/helpers/make-heading-slug-scope.ts

```diff
@@ -0,0 +1,31 @@
+import slugify from 'slugify'
+
+/**
+ * Returns a function that generates slugs from heading strings,
+ * while ensuring those slugs are unique across calls of the returned function
+ * by suffixing with a `-{number}` where needed.
+ */
+export function makeHeadingSlugScope(): (string) => string {
```

Comment:
Pretty excited about this, feels kinda basic in retrospect but it seems to work well ☺️ I don't _think_ there are any significant pitfalls to this approach (I was like "wait does this really work?" so added tests) but very open to changing course if there are 🙇

Reply:
Great idea using the closure! 🚀